### DISCUSSION TITLE
How to make react do only 1 render after calling zustand's set two time?

### DISCUSSION THREAD

**Discussion by mengbo-ji** [pmndrs/zustand#1958] [Q&A]
I want to call more actions from one action using Zustand.

I created something similar to this:

```
actionA() {
    set((state) =>
      produce(state, (draft) => {
        /* do changes */
      }),
    );
  },

  actionB() {
    set((state) =>
      produce(state, (draft) => {
        /* do changes */
      }),
    );
  },

  mainAction() {
    get().actionA();
    get().actionB();
  }
```
  
Both actionA and actionB are doing a lot of changes to the state.
actionB should work with the new state that was created by actionA
This is working fine, so I get the correct state at the end, but the problem is that after zustand's set react rerenders the components. I was expecting that only 1 render should happen similarly to when a setState called more times, so I was expecting some kind of 'batching'. Is not happening, I see the state (only for a very short time) that was created in actionA.

Is there any good practice how to do only 1 set in mainAction and keep the logic in actionA and actionB?

**Comment by mengbo-ji** [comment]
I need help~~~~~

**Reply by dai-shi** [reply]
It's batched: https://codesandbox.io/s/wizardly-breeze-jfcm8p?file=/src/App.js
